File: backend/amigos_java_backend/src/main/java/com/socialmedia/entity/Comment.java
package com.socialmedia.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "comments")
@Getter
@Setter
public class Comment extends BaseEntity {

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(name = "is_edited")
    private boolean isEdited = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Comment parent;

    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> replies = new ArrayList<>();
}

File: backend/amigos_dotnet_backend/Models/Entities/Comment.cs
using SocialMediaAdminBackend.Models.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SocialMediaAdminBackend.Models.Entities;

[Table("comments")]
public class Comment
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("post_id")]
    public long PostId { get; set; }

    [Column("user_id")]
    public long UserId { get; set; }

    [Column("content")]
    public string Content { get; set; } = string.Empty;

    [Column("parent_id")]
    public long? ParentId { get; set; }

    [Column("is_edited")]
    public bool IsEdited { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }

    [ForeignKey("PostId")]
    public Post? Post { get; set; }

    [ForeignKey("UserId")]
    public User? User { get; set; }
}